mediator/med_phases_prep_lnd_mod.F90

```diff
@@ -4,77 +4,15 @@ module med_phases_prep_lnd_mod
-  use med_methods_mod       , only : fldbun_getdata2d => med_methods_FB_getdata2d
-  use med_methods_mod       , only : field_getdata1d  => med_methods_Field_getdata1d
-  use med_methods_mod       , only : field_getdata2d  => med_methods_Field_getdata2d
-  use med_methods_mod       , only : State_GetScalar  => med_methods_State_GetScalar
-  use med_methods_mod       , only : State_SetScalar  => med_methods_State_SetScalar
-  use med_utils_mod         , only : chkerr           => med_utils_ChkErr
-  use med_constants_mod     , only : dbug_flag        => med_constants_dbug_flag
-  use med_internalstate_mod , only : InternalState, mastertask, logunit
-  use med_map_mod           , only : med_map_rh_is_created, med_map_routehandles_init
-  use med_map_mod           , only : med_map_field_packed, med_map_field_normalized, med_map_field
-  use med_merge_mod         , only : med_merge_auto
-  use glc_elevclass_mod     , only : glc_get_num_elevation_classes
-  use glc_elevclass_mod     , only : glc_mean_elevation_virtual
-  use glc_elevclass_mod     , only : glc_get_fractional_icecov
-  use perf_mod              , only : t_startf, t_stopf
+  use med_kind_mod, only : CX=>SHR_KIND_CX, CS=>SHR_KIND_CS, CL=>SHR_KIND_CL, R8=>SHR_KIND_R8
 
   implicit none
   private
 
   public  :: med_phases_prep_lnd
 
-  private :: map_glc2lnd_init
-  private :: map_glc2lnd
-
-  ! private module variables
-  character(len =*), parameter :: Sg_icemask = 'Sg_icemask'
-  character(len =*), parameter :: Sg_icemask_coupled_fluxes = 'Sg_icemask_coupled_fluxes'
-  character(len =*), parameter :: Sg_frac = 'Sg_ice_covered'
-  character(len =*), parameter :: Sg_frac_x_icemask = 'Sg_frac_times_icemask'
-  character(len =*), parameter :: Sg_topo = 'Sg_topo'
-  character(len =*), parameter :: Flgg_hflx = 'Flgg_hflx'
-
-  type, public :: ice_sheet_tolnd_type
-     character(CS)    :: name
-     logical          :: is_active
-     type(ESMF_Field) :: field_icemask_g           ! no elevation classes
-     type(ESMF_Field) :: field_frac_g_ec           ! elevation classes
-     type(ESMF_Field) :: field_frac_x_icemask_g_ec ! elevation classes
-     type(ESMF_Field) :: field_topo_x_icemask_g_ec ! elevation classes
-     type(ESMF_Mesh)  :: mesh_g
-  end type ice_sheet_tolnd_type
-  type(ice_sheet_tolnd_type) :: ice_sheet_tolnd(max_icesheets)
-
-  type(ESMF_field) :: field_icemask_l                ! no elevation classes
-  type(ESMF_Field) :: field_frac_l_ec                ! elevation classes
-  type(ESMF_Field) :: field_frac_x_icemask_l_ec      ! elevation classes
-  type(ESMF_Field) :: field_topo_x_icemask_l_ec      ! elevation classes
-
-  ! the number of elevation classes (excluding bare land) = ungriddedCount - 1
-  integer :: ungriddedCount ! this equals the number of elevation classes + 1 (for bare land)
-
-  logical :: glc2lnd_coupling = .false.
-  logical :: cism_evolve = .false.
+  real(r8), pointer :: dataptr_scalar_lnd(:,:)
```

Comment:
Do we need to assign them NULL?

Reply:
good idea - thanks for pointing this out.